stepwise-regression/script.whizzml

```diff
@@ -0,0 +1,99 @@
+;; A simple function to get the max value in a list
+(define (get-max xs) (reduce (lambda (x y) (if (> x y) x y)) (head xs) xs))
+
+;; Get feature names given ids
+(define (feature-names dataset-id ids)
+  (let (fields ((fetch dataset-id) "fields"))
+    (map (lambda (id) (fields [id "name"])) ids)))
+
+;; Create a dataset sample
+(define (sample-dataset ds-id rate oob)
+  (create-and-wait-dataset {"sample_rate" rate
+                            "origin_dataset"  ds-id
+                            "out_of_bag" oob
+                            "seed" "whizzml-example"}))
+
+;; Split a dataset into training and test sets
+(define (split-dataset ds-id rate)
+  (list (sample-dataset ds-id rate false)
+        (sample-dataset ds-id rate true)))
+
+;; Get the default set of input fields for this dataset (all preferred
+;; fields minus the objective field).
+(define (default-inputs dataset-id obj-id)
+  (let (fields ((fetch dataset-id) "fields")
+        fids (keys fields))
+    (filter (lambda (k) (and (fields [k "preferred"] false) (not (= obj-id k))))
+            fids)))
+
+;; Make a list of logistic regressions using the given dataset,
+;; objective field and selected features.  For each lr, add one of the
+;; potential features in potentials to the input feature list.
+
+(define (make-lr dataset-id obj-field selected potentials)
+  (let (lr-req {"dataset" dataset-id "objective_field" obj-field}
+        make-req (lambda (fid)
+                   (assoc lr-req "input_fields" (cons fid selected)))
+        all-reqs (map make-req potentials))
```

Comment:
just in case you like it better, you could also compute `all-reqs` with a `for`:

```
all-reqs (for (fid potentials)
           {"dataset" dataset-id 
            "objective_field" obj-field 
            "input_fields" (cons fid selected)})
```

(i find it more readable, but feel free to ignore)